test/deploy.js

```diff
@@ -0,0 +1,98 @@
+const test = require('ava');
+const sinon = require('sinon');
+const proxyquire = require('proxyquire').noCallThru();
+const { promisify } = require('util');
+const { exec } = require('child_process');
+
+const execAsync = promisify(exec);
+
+const CLI = 'node dist/cli.js';
+
+test('deploy help', async t => {
+  const output = (await execAsync(`${CLI} deploy --help`)).stdout;
+  t.snapshot(output);
+});
+
+test('deploy no args', async t => {
+  const error = await t.throwsAsync(execAsync(`${CLI} deploy`));
+  t.true(error.message.includes('Missing required option: --contractName'));
+});
+
+const TX_HASH = '0x1';
+const DEPLOYMENT_ID = 'abc';
+const ADDRESS = '0x2';
+const FAKE_CHAIN_ID = '1';
+
+function setupFakeDefender(t) {
+  const fakeDefenderClient = {
+    deployContract: () => {
+      return {
+        txHash: TX_HASH,
+        deploymentId: DEPLOYMENT_ID,
+        address: ADDRESS,
+        status: 'completed'
+      };
+    },
+    getDeployedContract: () => {
+      return {
+        txHash: TX_HASH,
+        deploymentId: DEPLOYMENT_ID,
+        address: ADDRESS,
+        status: 'completed'
+      };
+    }
+  };
+  t.context.spy = sinon.spy(fakeDefenderClient, 'deployContract');
+
+  t.context.command = proxyquire('../dist/commands/deploy', {
+    '../internal/client': {
+      getDeployClient: () => fakeDefenderClient,
+    }
+  });
```

Comment:
Not sure if `proxyquire` is completely necessary but seems like the dependency hasn't been updated in a while, which may need us to remove it afterwards if there's a security advisory.

I wonder if we can use only [a Sinon stub](https://sinonjs.org/releases/v17/stubs/) for this since it allows for overriding the behavior.